### PR TITLE
Last tweaks to the tools and JS envs

### PR DESCRIPTION
**Pull request by sjrd**

### CODE REVIEW

**Comment by sjrd** [comment]
Review by @gzm0

**Comment by sjrd** [comment]
With these changes, the generated Scaladocs of the Tools and the JS Envs actually make sense :p

File: tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/CoreJSLibs.scala [hunk 1]
```diff
@@ -20,7 +20,8 @@ import org.scalajs.core.tools.linker.backend.OutputMode
 import scala.collection.immutable.Seq
 import scala.collection.mutable
 
-object CoreJSLibs {
+// The only reason this is not private[backend] is that Rhino needs it
```
Comment (gzm0) [inline tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/CoreJSLibs.scala]: `private[emitter]`?

**Comment by gzm0** [comment]
One thing I'm a bit weary of, is that we privilege ourselves to use deep internals of the linking pipeline to implement one of the jsEnvs. Maybe it would be worth leaving a comment on one (or some) of these `private[scalajs]` entry points saying something the like: "If you would like to implement a lazy-loading JSEnv (like RhinoJSEnv) and have reached this point, **don't** move your package or parts thereof to `scalajs`, but contact us and we'll spec and create a clean interface to do so.".

In any case, I completely agree that what we have in this PR is sufficient for the moment (i.e. no need to do fancy interfacing for lazy loading etc. at this point).

**Comment by gzm0** [comment]
That's all.

File: tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/CoreJSLibs.scala (hunk 1 above)
Comment (sjrd) [inline tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/CoreJSLibs.scala]: It is also needed by the Closure backend

**Comment by sjrd** [comment]
Yes, ideally a clean interface would be needed. It will be easier to add one later, now that everything is private, when we need it. 

Honestly I don't think anyone else but us will ever write a lazy loading env, so that issue will probably not arise anytime soon.

**Comment by sjrd** [comment]
@gzm0 I'm not sure whether you want me to do anything on this PR?

File: tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/CoreJSLibs.scala (hunk 1 above)
Comment (gzm0) [inline tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/CoreJSLibs.scala]: Ah, ok.

**Comment by gzm0** [comment]
> I'm not sure whether you want me to do anything on this PR?

Nah, let's just leave it.

**Comment by gzm0** [comment]
LGTM

**Pull request merged**